pkg/runtime/setup.go

```diff
@@ -267,10 +277,24 @@ func (s *setup) onArtifactBuildReady(blog *buildlog.BuildLog, artifact *buildpla
 }
 
 func (s *setup) obtain(artifact *buildplan.Artifact) (rerr error) {
-	// Download artifact
-	b, err := s.download(artifact)
-	if err != nil {
-		return errs.Wrap(err, "download failed")
+	var b []byte
+	if s.opts.FromArchiveDir == "" {
+		// Download artifact
+		var err error
+		b, err = s.download(artifact)
+		if err != nil {
+			return errs.Wrap(err, "download failed")
+		}
+	} else {
+		// Read the artifact from the archive.
+		var err error
+		name := artifact.ArtifactID.String() + s.opts.ArtifactExt
+		artifactFile := filepath.Join(s.opts.FromArchiveDir, name)
+		logging.Debug("Reading file '%s' for '%s'", artifactFile, artifact.DisplayName)
+		b, err = fileutils.ReadFile(artifactFile)
+		if err != nil {
+			return errs.Wrap(err, "read from archive failed")
+		}
```

Comment:
I think since this is bypassing the event reporting of downloads we will have a progress indicator for downloads that never moves. 

We'll want to ensure that those events are still send, but we probably also want to update our runbit logic so that instead of "Downloading" we say "Unpacking" when using the archive approach.

Reply:
Okay, I'll add the download events. In testing, this step went by so fast I didn't even see a download bar. I only saw a brief "Unpacking" bar since setting up from an archive does go through this step.